### ISSUE TITLE
Caption formatting not rendering when using knitr::include_graphics in pdf

### ISSUE THREAD

**Issue by Athanasiamo** [rstudio/rmarkdown#1730]
I've been struggling for a while with a document that has caption formatting that fails at some point. I have finally been able to figure out when this is happening. When I use `knitr::include_graphics` the caption formatting no longer is applied. 

I have a document where I need to use orca for plotly captures, and so I need to be using include_graphics to add them to the document. I was unsure whether this was a rmarkdown or knitr issue, though. 

Here is a MRE of it happening in an Rmd

---
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r pressure, echo=FALSE, fig.cap="some **formatted** text"}
plot(pressure)
```

```{r pressure2, echo=FALSE, fig.cap="some **formatted** text"}
png(filename="pressure.png")
plot(pressure)
dev.off()

knitr::include_graphics("pressure.png")
```
 
```{r pressure3, echo=FALSE, fig.cap="some **formatted** text"}
library(knitr)
include_graphics("pressure.png")
```

**Comment by cderv** [comment]
Hi, 

thanks for the report. I believe the formatting of the GH issue made the chunk options disappeared and it is not easy to reproduce. I guess you have a `fig.cap` somewhere... 🤔 
Please follow advice here for better formatting
https://yihui.org/issue/#please-format-your-issue-correctly

Thank you !

**Comment by Athanasiamo** [comment]
Hi.

Sorry about that.

Let me try again

````
---
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Formatting working on these fig caps

```{r pressure, fig.cap='Some **formatted** text', fig.width=4, fig.height=3}
# Fig 1
plot(pressure)
```

```{r pressure2, fig.width=3, fig.cap='Some **formatted** text'}
# Fig 2
plot(pressure)
# Fig 3
plot(faithful)
```

# Fig cap omitted on this one
```{r pressure5, fig.show='hold', fig.width=3, fig.cap='Some **formatted** text'}
# unnumbered figure, side-by-side
plot(pressure)
plot(faithful)
```

# Formatting lost on these
```{r pressure6, out.width='100%', fig.cap='Some **formatted** text'}
# Fig 4
plot(pressure)
```

```{r pressure4, fig.cap='Some **formatted** text'}
# Fig 5
png(filename="pressure.png")
plot(pressure)
dev.off()
knitr::include_graphics("pressure.png")
```
````

**Comment by Athanasiamo** [comment]
Hi! I made progress in my manuscript using the reference labelling for captions as explained
[here](https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html#text-references). Sorry I didn't try this before, I must have missed this chapter.

 But I can't seem to get it to work in the example I made above. Likely because it only works in bookdown::pdf_book ?

````

---
output: bookdown::pdf_book
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Formatting working on these fig caps

```{r pressure, fig.cap='Some **formatted** text', fig.width=4, fig.height=3}
# Fig 1
plot(pressure)
```

```{r pressure2, fig.width=3, fig.cap='Some **formatted** text'}
# Fig 2
plot(pressure)
# Fig 3
plot(faithful)
```

# Fig cap omitted on this one
```{r pressure5, fig.show='hold', fig.width=3, fig.cap='Some **formatted** text'}
# unnumbered figure, side-by-side
plot(pressure)
plot(faithful)
```

# Formatting lost on these
```{r pressure6, out.width='100%', fig.cap='Some **formatted** text'}
# Fig 4
plot(pressure)
```

```{r pressure4, fig.cap='Some **formatted** text'}
# Fig 5
png(filename="pressure.png")
plot(pressure)
dev.off()
knitr::include_graphics("pressure.png")
```

## But works when doing the ref-version of captioning

(ref:pressure7-cap) Some **formatted** text

```{r pressure7, out.width='100%', fig.cap='(ref:pressure7-cap)'}
# Fig 4
plot(pressure)
```

(ref:pressure8-cap) Some **formatted** text

```{r pressure8, fig.cap='(ref:pressure8-cap)'}
# Fig 5
png(filename="pressure.png")
plot(pressure)
dev.off()
knitr::include_graphics("pressure.png")
```


````

**Comment by cderv** [comment]
If you need bookdown feature in a single document, I think you need to use `bookdown::pdf_document2`
https://bookdown.org/yihui/bookdown/a-single-document.html

**Comment by cderv** [comment]
About the formatting of caption in your first example, this is due to how knitr is handling the figure inclusion. In some case, it will add figures in the md document as markdown like that `![md formatted label](path/to/image)`. In other case, to handle specific options, it will include it as raw html or latex depending of the output format. For example with `knitr::include_graphics`
```md
\begin{figure}
\includegraphics[width=6.67in]{pressure} \caption{Some **formatted** text}\label{fig:pressure4}
\end{figure}
```

In the former case, I think pandoc will parse as markdwn and format, in the latter, I think it will parse as latex and do not format the label.

Using `fig.cap='Some \\textbf{formatted} text'` in the pressure4 chunk will produce a formatted label in latex. However, it is difficult to know when the chunk will produce markdown or latex for image links. 🤔 

Formating outside using ref label is a good idea. For that you'll need to use the `bookdown::pdf_document2` format. You example above seems to work for me with this format.

Hope it helps.

**Comment by Athanasiamo** [comment]
Nice explanation, and also nice to know that this was a little more complicated than I thought. Doing it outside really helped, and it is an rticles doc, and using another doc type. 

It's working now, so I'm happy to have learned something new!

**Comment by adakteen** [comment]
Hi there,

I'm having a very similar issue, but the problem is handled differently between my Mac and Windows machines. On Windows, the figure captions are rendered just as I want and the *.png files get sized correctly. However, the exact same code on Mac produces the errors explained above.
Can anyone please explain why this is the case?
Kind regards,

**Comment by cderv** [comment]
Hi @adakteen , 

It is not easy to help you on the basis of an old issue. For questions you could ask on RStudio community (https://community.rstudio.com/c/R-Markdown/10) as it is better suited for this. If you found an issue I would suggest open a new one (linking to this one if you think it is related) 

It is really not easy to help you without a reproducible example of what you are trying to do. Every case could be different and when something is different between OS it could come from the environment you are working on. 

Thank you!
